src/Microsoft.Identity.Web/Microsoft.Identity.Web.csproj

```diff
@@ -40,13 +46,18 @@
   
   <PropertyGroup>
     <TargetFramework>netcoreapp3.1</TargetFramework>
+    <SignAssembly>true</SignAssembly>
+    <AssemblyOriginatorKeyFile>../../build/MSAL.snk</AssemblyOriginatorKeyFile>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <!-- The MSAL.snk has both private and public keys -->
```

Comment:
Should we check in only public though?

Reply:
No, it's fine. Do what MSAL does. And you don't need this property either, delaySign is false by default I think.

Assembly signing is NOT a security mechanism. Our Release build will add a proper digital signature. The only reason we do this "sign assembly" is to make sure the DLL is `strongly named`. I believe that in the past .net folks wanted this to be a security measure as well, but it doesn't hold up. The strong naming is good though. So current guidance is to commit an snk file that contains both public and private keys. The goal is to have contributors that are able to create strongly-signing DLLs, otherwise they can't test their changes.

Reply:
Makes sense. Thanks. 👍

Reply:
thanks bogdan. :)

---
In reply to: [402375802](https://github.com/AzureAD/microsoft-identity-web/pull/75#discussion_r402375802) [](ancestors = 402375802)